syntax = "proto3";

package envoy.extensions.xds;

import "envoy/config/common/key_value/v3/config.proto";
import "udpa/annotations/status.proto";
import "validate/validate.proto";

option java_package = "io.envoyproxy.envoy.extensions.xds";
option java_outer_classname = "KeyValueStoreXdsDelegateConfigProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/xds";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#extension: envoy.xds_delegates.kv_store]
//
// Configuration for a KeyValueStore-based XdsResourcesDelegate implementation.
//
// [#not-implemented-hide:]
message KeyValueStoreXdsDelegateConfig {
  // Configuration for the KeyValueStore that holds the xDS resources.
  config.common.key_value.v3.KeyValueStoreConfig key_value_store_config = 1;
};
